README.md

```diff
@@ -5,7 +5,126 @@
+
+### What does the Workleap.DomainEventPropagation package do?
+* Provides list of existing Topics
+* Serializes domain events
+* Manages publication fields
+* Provides webhook endpoint (.net core only)
+* Provides IDomainEvent and IDomainEventHandler<> interfaces
+* Deserializes domain events
+* Mediates call to DomainEventHandlers
+* Validates configurations
+* Handles topic subscriptions
+
+### Using the Workleap.DomainEventPropagation.Publishing package to publish events
+
+When using dotnet core, you can register event propagation at startup in the service collection.
+```
+// Method 1: Lazily bind the options to a configuration section
+services.AddEventPropagationPublisher();
+services.AddOptions<EventPropagationPublisherOptions>().BindConfiguration(EventPropagationPublisherOptions.SectionName);
+
+// Method 2: Set options values directly in C#
+services.AddEventPropagationPublisher(opt =>
+{
+  opt.TopicName = "<topic_name_to _publish_to>",
+  opt.TopicAccessKey = "<provided from keyVault>",
+  opt.TopicEndpoint = "<azure_topic_uri>"
+});
+```
+Configuration is required. Configuration can be loaded from the appsettings file by passing the IConfiguration instance (see above). The topic access key should be stored securely in a key vault.
+```json
+// appsettings.json
+{
+  "EventPropagation": {
+    "TopicName": "<topic_name_to_publish_to>",
+    "TopicAccessKey": "<keyVault_provided_value>",
+    "TopicEndpoint": "<azure_topic_uri>"
+  }
+}
+```
+To publish an event, use the IEventPropagationClient interface (via injection). Use the PublishDomainEventAsync to publish the event. The required Subject field is a string description to provide context for the event.
+
+```
+var domainEvent = new ExampleDomainEvent
+{
+    Id = Guid.NewGuid().ToString(),
+    Date = DateTime.UtcNow
+};
+
+await this._eventPropagationClient.PublishDomainEventAsync(subject: "TestEventPublication", coolCandorEvent);
+```
+
+#### Restrictions
+* Your service can publish events on only 1 topic.
+* The topic the service publishes to must be in the Topics class provided by the Workleap.EventPropagation package.
+
+### Using the Workleap.EventPropagation package to subscribe to events
+When using dotnet core, you can register event propagation subscriptions at startup in the service collection.  To configure the subscriber, the list of subscribed topics is required.
+
+```
+services
+    .AddEventPropagationSubscriber()
+    .AddDomainEventHandlersFromAssembly(Assembly.GetExecutingAssembly())
+    .Configure(new[] { Topics.Organisation, Topics.Candor });
+
+// or add a single domain event handler
+.AddDomainEventHandler<ExampleDomainEventHandler>()
+```
+An API endpoint must be made available to receive events from the EventGrid topic. The package provides this endpoint for dotnet core projects. It can be added like so.
```

Comment:
```suggestion
An API endpoint must be made available to receive events from the Event Grid topic. The package provides this endpoint for ASP.NET Core projects. It can be added like so:
```